### PR TITLE
Integration test using docker compose

### PR DESCRIPTION
**Pull request by renan-souza**
- Integration test (defined in .github/workflows/test.yml) works calling the script `imagemagick-files.py` from agent2 communicating with agent1, using the current compose file.

Tests passing: https://github.com/ORNL/zambeze/actions/runs/3888182470

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Integration Test
```
Comment (JoshuaSBrown) [inline .github/workflows/test.yml:1]: Let's change the name so it is clear what the action is doing. i.e. Runs Rsync Example as Integration Test.

File: tests/test_imagemagick_files.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    # Step 4 launch example
+    process = subprocess.Popen(
+        command,
+        shell=False,
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.STDOUT)
+
+    stdout, stderr = process.communicate()
+    print(stdout)
+    print(stderr)
```
Comment (JoshuaSBrown) [inline tests/test_imagemagick_files.py:40]: Maybe just clean this file of the extraneous print statements.

File: deployment/compose.yml [hunk 3]
```diff
@@ -10,24 +11,28 @@ services:
...
       NATS_HOST: nats
       NATS_PORT: 4222
-
-  zambeze_agent2:
+      PYTHONWARNINGS: "ignore::DeprecationWarning"
     container_name: zambeze2
     image: zambeze/zambeze:latest
-    tty: true
-    stdin_open: true
-    volumes:
```
Comment (JoshuaSBrown) [inline deployment/compose.yml:27]: So this was the culprit, I don't know how you figured this out.

File: deployment/compose.yml [hunk 4]
```diff
@@ -10,24 +11,28 @@ services:
...
+      PYTHONWARNINGS: "ignore::DeprecationWarning"
     container_name: zambeze2
     image: zambeze/zambeze:latest
-    tty: true
-    stdin_open: true
-    volumes: 
+    volumes:
       - ../db:/zambeze/db
-      - ../:/zambeze  # This line is needed only for development.
+      - ../:${CONTAINER_VOL_MNT_POINT:-/zambeze}  # This line is needed only to facilitate development.
```
Comment (JoshuaSBrown) [inline deployment/compose.yml:33]: Where in the CI is it set?

**Review by JoshuaSBrown — APPROVED** [review]
If it works, I'm happy to get this in as soon as possible, thanks for cleaning it up. I have a few minor suggestions. I would also like to know more about the details of the problem so I don't reproduce it. Thanks again this was a lot of work and much appreciated.

**Comment by JoshuaSBrown** [comment]
Please close the other PR that is opened against main when you have merged this.

File: deployment/compose.yml (hunk 4 above)
Comment (renan-souza) [reply deployment/compose.yml:33]: It's set in the test.yml (I renamed this yml file). I changed the #comments to make it clear.

**Pull request merged**